services-marionette/firefox_services_tests/apps/fxa/login.py

```diff
@@ -1,45 +1,42 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
-from firefox_services_tests.apps.base import Base
-from firefox_services_tests.helpers import utils
-
+from helpers import utils
 from marionette_driver import By
 
+from apps.base import Base
+
 
 class LoginPage(Base):
     _fxa_email_input_locator = (By.CSS_SELECTOR, 'input.email')
     _fxa_password_input_locator = (By.ID, 'password')
-    _fxa_sign_in_button_locator = (By.ID, 'submit-btn')
+    _fxa_sign_up_button_locator = (By.ID, 'submit-btn')
     _fxa_sign_in_header_locator = (By.ID, 'fxa-signin-header')
     _fxa_sign_up_header_locator = (By.ID, 'fxa-signup-header')
     _fxa_age_input_locator = (By.ID, 'age')
     _fxa_create_account_link_locator = (By.CSS_SELECTOR, 'a.sign-up')
     _fxa_accept_request_permission_button_locator = (By.ID, 'accept')
     _fxa_signup_complete_header_locator = (By.ID, 'fxa-signup-complete-header')
-    _pocket_start_saving_button_locator = (
-        By.XPATH, '//a[text()="Start Saving"]')
-    _pocket_logo_header_locator = (By.CSS_SELECTOR, 'h1.pocket_logo')
 
     def __init__(self, marionette):
         Base.__init__(self, marionette)
 
-    def login_to_fxa(self):
-        self.wait_for_element_displayed(*self._fxa_sign_in_header_locator)
-        # TODO: We need to add or implement functionality that pulls credentials a manifests  file
-        self.marionette.find_element(*self._fxa_email_input_locator)\
-            .send_keys()
-        self.marionette.find_element(*self._fxa_password_input_locator)\
-            .send_keys()
-        self.click_element(*self._fxa_sign_in_button_locator).click()
-        self.wait_for_element_displayed()
+    # def login_to_fxa(self):
+    #     self.wait_for_element_displayed(*self._fxa_sign_in_header_locator)
+    #     # TODO: We need to add or implement functionality that pulls credentials a manifests  file
+    #     self.marionette.find_element(*self._fxa_email_input_locator)\
+    #         .send_keys()
+    #     self.marionette.find_element(*self._fxa_password_input_locator)\
+    #         .send_keys()
+    #     self.click_element(*self._fxa_sign_in_button_locator).click()
+    #     self.wait_for_element_displayed()
+
+    def sign_up_for_fxa(self, start_at_signin, request_permission):
+        if start_at_signin:
+            self.wait_for_element_displayed(*self._fxa_sign_in_header_locator)
```

Comment:
Keep in mind that this will not allow you to cache element instances. Instead an instance of the given DOM element will be created each time!